### PR TITLE
Add custom Show references permission and apply it to the view and toolbar button

### PR DESCRIPTION
**Pull request by chaosk**

### CODE REVIEW

File: djangocms_references/cms_toolbars.py [hunk 1]
```diff
@@ -13,16 +15,19 @@ class ReferencesToolbar(CMSToolbar):
     """
 
     def populate(self):
+        opts = References._meta
+        if not self.request.user.has_perm(
```
Comment (monikasulik) [inline djangocms_references/cms_toolbars.py]: could be `if not ... or obj is None` cause the same thing happens on both ifs.

File: djangocms_references/models.py [hunk 2]
```diff
@@ -0,0 +1,13 @@
+from django.db import models
+from django.utils.translation import ugettext_lazy as _
+
+
+class References(models.Model):
+    """Dummy model used for permission handling. No DB tables are
+    actually created.
+    """
```
Comment (monikasulik) [inline djangocms_references/models.py]: You sure this is preferable to a data migration? Cause I don't much like the idea of a model class existing (even with `managed = False`) solely so we can populate some data in a different table.

Comment (chaosk) [reply djangocms_references/models.py]: I think it's more clean than creating a Permission that's detached from any model. Besides, a Permission has to be attached to a ContentType, so we'd have to create a dummy ContentType anyway?

File: djangocms_references/cms_toolbars.py (hunk 1 above)
Comment (chaosk) [reply djangocms_references/cms_toolbars.py]: :+1:

**Pull request merged**